egs2/TEMPLATE/svs1/pyscripts/utils/prep_segments_from_xml.py

```diff
@@ -0,0 +1,179 @@
+#!/root/anaconda3/bin/python3
+import argparse
+import math
+import os
+import sys
+import music21 as m21
+
+"""Divide songs into segments according to structured musicXML."""
+
+class LabelInfo(object):
+    def __init__(self, start, end, label_id, isNote, note):
+        self.label_id = label_id
+        self.start = start
+        self.end = end
+        self.isNote = isNote
+        self.note = note
+    
+    def extend(self, start, end, label_id):
+        if self.start == self.end:
+            self.label_id = label_id
+            self.start = start
+            self.end = end
+        else:
+            self.end = end
+
+
+class SegInfo(object):
+    def __init__(self):
+        self.segs = []
+        self.start = -1
+        self.end = -1
+        self.off = -1
+
+    def add(self, start, end, label, note):
+        start = float(start)
+        end = float(end)
+        if self.start < 0 or self.start > start:
+            self.start = start
+            self.off = note.offset
+        if self.end < end:
+            self.end = end
+        note.offset -= self.off
+        self.segs.append((start, end, label, note))
+
+    def split(self, threshold=10):
+        seg_num = math.ceil((self.end - self.start) / threshold)
+        if seg_num == 1:
+            return [self.segs]
+        avg = (self.end - self.start) / seg_num
+        return_seg = []
+
+        start_time = self.start
+        cache_seg = []
+        off = self.segs[0][3].offset
+        for seg in self.segs:
+            cache_time = seg[1] - start_time
+            if cache_time > avg:
+                return_seg.append(cache_seg)
+                start_time = seg[0]
+                off = seg[3].offset
+                seg[3].offset -= off
+                cache_seg = [seg]
+            else:
+                seg[3].offset -= off
+                cache_seg.append(seg)
+
+        return_seg.append(cache_seg)
+        return return_seg
+
+
+def pack_zero(file_id, number, length=4):
+    number = str(number)
+    return file_id + "_" + "0" * (length - len(number)) + number
+
+
+def get_parser():
+    parser = argparse.ArgumentParser(
+        description="Prepare segments from HTS-style alignment files",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument("scp", type=str, help="scp folder")
+    parser.add_argument(
+        "threshold", type=int, help="threshold for silence identification."
+    )
+    parser.add_argument(
+        "--silence", action="append", help="silence_phone", default=["pau"]
+    )
+    parser.add_argument(
+        "--xml_dump", type=str, default="xml_dump", help="xml dump directory")
```

Comment:
we may simply follow the espnet dump directory convention. But you can use the current way for now, we can easily change it later